homeassistant/components/bluetooth/__init__.py

```diff
@@ -212,7 +199,40 @@ def _async_trigger_discovery() -> None:
         EVENT_HOMEASSISTANT_STOP, hass_callback(lambda event: cancel())
     )
 
+
+async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
+    """Set up the bluetooth integration."""
+    bluetooth_adapters = get_adapters()
+    bluetooth_storage = BluetoothStorage(hass)
+    slot_manager = BleakSlotManager()
+    integration_matcher = IntegrationMatcher(await async_get_bluetooth(hass))
+
+    slot_manager_setup_task = hass.async_create_task(
+        slot_manager.async_setup(), "slot_manager setup"
+    )
+    processor_setup_task = hass.async_create_task(
+        passive_update_processor.async_setup(hass), "passive_update_processor setup"
+    )
+    storage_setup_task = hass.async_create_task(
+        bluetooth_storage.async_setup(), "bluetooth storage setup"
+    )
+    integration_matcher.async_setup()
+    manager = HomeAssistantBluetoothManager(
+        hass, integration_matcher, bluetooth_adapters, bluetooth_storage, slot_manager
+    )
+    set_manager(manager)
+
+    await storage_setup_task
+    await manager.async_setup()
+    hass.data[DATA_MANAGER] = models.MANAGER = manager
+
+    hass.async_create_background_task(
+        _async_start_adapter_discovery(hass, manager, bluetooth_adapters),
+        "start_adapter_discovery",
+    )
+    await slot_manager_setup_task
```

Comment:
If `storage_setup_task` raises, we will never await this task to retrieve the exception.

Reply:
Its awaited on https://github.com/home-assistant/core/pull/111304/files#diff-574d610a162f0473fdbb7e6e91b25d31d42fbb35329c95397b49a58eebd1816dR225